File: Python/백준/Silver/10773. 제로/제로.py
k = int(input())
g =[]
for i in range(k):
    c = int(input())
    if c == 0:
        g.pop()
    else:
        # 해당 수를 쓴다
        g.append(c)  
A = 0
for i in g:
    A += i
print(A)

File: Java/백준/Silver/10773. 제로/제로.java
import java.util.Stack;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        Stack<Integer> list = new Stack<>();

        for (int i = 0; i < n; i++){
            int num = input.nextInt();
            if(num == 0){
                list.pop();
            }
            else{
                list.push(num);
            }
        }

        int result = 0;
        for (int i : list){
            result += i;
        }
        System.out.println(result);
    }
}
